docs/guides/tfhers.md

```diff
@@ -0,0 +1,341 @@
+# TFHE-rs Compatibility
+
+This document explains how to combine Concrete and [TFHE-rs](https://github.com/zama-ai/tfhe-rs) computations together. This allows users to convert ciphertexts from Concrete to TFHE-rs, and vice versa. This allows users to run a computation with both libraries without requiring a decryption.
+
+## Table of Content
+
+- [The Basics](#the-basics)
+- [First Use Case](#first-use-case-shared-secret-key)
+- [Second Use Case](#second-use-case-pregenerated-tfhe-rs-keys)
+- [Serialization of Ciphertexts and Keys](#serialization-of-ciphertexts-and-keys)
+
+## The Basics
+
+There are differences between Concrete and TFHE-rs, so ensuring compatibility between them involves more than just data serialization. To achieve compatibility, we need to consider two main aspects.
+
+1. In TFHE-rs and Concrete, integers are encoded differently. Both use LWE ciphertexts, but organized differently. In Concrete, TFHE-rs `FheInt/FheUint` are represented as a vector of `n` integers. `n` is determined by dividing the TFHE-rs integer bit-width by the TFHE-rs message bit-width. These bit-width values, along with the carry bit-width, are the encoding parameters used in Concrete to represent TFHE-rs big integers. When working with a TFHE-rs integer type in Concrete, you can use the `.encode(...)` and `.decode(...)` functions to see this in practice.
```

Comment:
I feel like it's too much for this guide. I see other guides being solely focused on the usage, rather than on the how it works underneath. This guide is already going into some necessary crypto concepts, but I don't think this one is needed to understand how to use the feature.

Reply:
ok let @yuxizama decide